flocker/node/agents/test/test_blockdevice.py

```diff
@@ -698,6 +736,17 @@ def add_application_with_volume(node_state):
                                   mountpoint=FilePath(b"/data")))})
 
 
+def _call_calculate_changes(deployer, cluster_configuration, cluster_state,
+                            local_state=None):
+    """
+    Wrapper for deployer.calculate_changes with sane defaults for testing.
+    """
+    if local_state is None:
+        local_state = EMPTY_LOCAL_STATE
```

Comment:
Doesn't seem like this is the right default, especially once we have code that actually uses it. Perhaps as a minimum it should be extracting info from cluster_state?